poi-ooxml/src/main/java/org/apache/poi/xssf/usermodel/XSSFOptimiser.java

```diff
@@ -0,0 +1,247 @@
+     * Uses font PROPERTIES (not index) so styles with identical but separate fonts
+     * are detected as duplicates.
+     */
+    private static String getStyleSignature(XSSFCellStyle style) {
+        StringBuilder sb = new StringBuilder();
+        // Use font properties, not index, so identical fonts are detected
+        XSSFFont font = style.getFont();
+        sb.append("font:").append(getFontSignature(font)).append(";");
+        sb.append("fmt:").append(style.getDataFormat()).append(";");
+        sb.append("align:").append(style.getAlignment()).append(";");
+        sb.append("valign:").append(style.getVerticalAlignment()).append(";");
+        sb.append("wrap:").append(style.getWrapText()).append(";");
+        sb.append("rotation:").append(style.getRotation()).append(";");
+        sb.append("indent:").append(style.getIndention()).append(";");
+        sb.append("borderL:").append(style.getBorderLeft()).append(";");
+        sb.append("borderR:").append(style.getBorderRight()).append(";");
+        sb.append("borderT:").append(style.getBorderTop()).append(";");
+        sb.append("borderB:").append(style.getBorderBottom()).append(";");
+        sb.append("fillFg:").append(colorToString(style.getFillForegroundColorColor())).append(";");
+        sb.append("fillBg:").append(colorToString(style.getFillBackgroundColorColor())).append(";");
+        sb.append("fillPat:").append(style.getFillPattern()).append(";");
+        sb.append("locked:").append(style.getLocked()).append(";");
+        sb.append("hidden:").append(style.getHidden()).append(";");
+        return sb.toString();
+    }
+
+    /**
+     * Generate a signature string for a font to detect duplicates.
+     */
+    private static String getFontSignature(XSSFFont font) {
+        StringBuilder sb = new StringBuilder();
+        sb.append("name:").append(font.getFontName()).append(";");
+        sb.append("size:").append(font.getFontHeightInPoints()).append(";");
+        sb.append("bold:").append(font.getBold()).append(";");
+        sb.append("italic:").append(font.getItalic()).append(";");
+        sb.append("strike:").append(font.getStrikeout()).append(";");
+        sb.append("underline:").append(font.getUnderline()).append(";");
+        sb.append("color:").append(colorToString(font.getXSSFColor())).append(";");
+        sb.append("typeOffset:").append(font.getTypeOffset()).append(";");
+        sb.append("charset:").append(font.getCharSet()).append(";");
+        return sb.toString();
+    }
+
+    /**
+     * Convert a color to a string for comparison.
+     */
+    private static String colorToString(Color color) {
+        if (color == null) {
+            return "null";
+        }
+        if (color instanceof XSSFColor) {
+            XSSFColor xssfColor = (XSSFColor) color;
+            byte[] rgb = xssfColor.getRGB();
+            if (rgb != null) {
+                return String.format("#%02X%02X%02X", rgb[0] & 0xFF, rgb[1] & 0xFF, rgb[2] & 0xFF);
```

Comment:
forbidden api call - you need to use the version of string.format that takes a Locale as an input.

Locale.Root is fine

Reply:
Done